bot/monkey_patches.py

```diff
@@ -50,3 +52,36 @@ async def honeybadger_type(self, channel_id: int) -> None:  # noqa: ANN001
             pass
 
     http.HTTPClient.send_typing = honeybadger_type
+
+
+def patch_message_converter() -> None:
+    """
+    Make the Message converter infer channelID from the given context.
+
+    Discord.py's Message converter is supposed to infer channelID based
+    on ctx.channel if only a messageID is given. A "refactor" (linked below)
+    a few weeks before d.py's archival broke this, so that if only a messageID
+    is given to the converter, it will only find that message if it's in the bot's
+    cache.
+
+    https://github.com/Rapptz/discord.py/commit/1a4e73d59932cdbe7bf2c281f25e32529fc7ae1f
+    """
+    log.debug("Patching Message converter to infer channelID from context.")
+    original = commands.PartialMessageConverter._get_id_matches
+
+    def _get_as_snowflake(data: t.Any, key: str) -> t.Optional[int]:
+        try:
+            value = data[key]
+        except KeyError:
+            return None
+        else:
+            return value and int(value)
```

Comment:
this is identical to what we're mocking, so I don't understand the point?

Reply:
No? We're mocking a function which calls this in the middle because we want to have slightly different behavior (see the side effect on mock). We duplicated the base function because we need to call it in the side effect, and this was the easiest way.

Reply:
Since this function is the one we're mocking, and within the mock we calling it first, before inferring, we can't have the mock side effect call the function it's mocking, so we copied the function instead.